### PR TITLE
Fix `onnx.Gather` conversion to `stablehlo` in dynamic case

### PR DESCRIPTION
**Pull request by srcarroll**
The previous implementation of `ONNXGatherOpLoweringToStablehlo` didn't handle dynamic input cases correctly yielding an invalid `stablehlo.dynamic_broacast_in_dim` op.  Nor was the dynamic case even tested. This patch fixes this and adds a test.

### CODE REVIEW

File: src/Conversion/ONNXToStablehlo/Tensor/Gather.cpp [hunk 1]
```diff
@@ -56,19 +57,16 @@ struct ONNXGatherOpLoweringToStablehlo : public ConversionPattern {
...
+    if (!inputType.isDynamicDim(axisLit)) {
       int64_t axisDimSizeLit = inputType.getShape()[axisLit];
       axisDimSize = getShapedInt(loc, rewriter, axisDimSizeLit, indices);
     } else {
-      Value inputShape = rewriter.create<shape::ShapeOfOp>(loc, data);
-      Value indicesShape = rewriter.create<shape::ShapeOfOp>(loc, indices);
-      Value axisDimSizeIndexValue =
-          rewriter.create<shape::GetExtentOp>(loc, inputShape, axisLit);
-      Value axisDimSizeValue = rewriter.create<arith::IndexCastOp>(
-          loc, indicesType.getElementType(), axisDimSizeIndexValue);
```
Comment (srcarroll) [inline src/Conversion/ONNXToStablehlo/Tensor/Gather.cpp:68]: the issue was that `axisDimSizeValue` is an `i64` not a `tensor<i64>` i could just use the old method but add a `tensor.from_elements` to get the right type. but i thought `tensor.dim` + `tensor.splat` was more straight forward. however if this is against design principles I can do the former.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@Connor-XY could you help take a look at this patch, if you don't mind? thanks.

**Review by Connor-XY — APPROVED** [review]
LGTM! I really appreciate that you fix the bug and add a test for dynamic case.

**Comment by tungld** [comment]
Thank you very much @Connor-XY and @srcarrol!

**Review by tungld — APPROVED** [review]
LGTM.

**Pull request merged**